### PR TITLE
Constrictor 5.0.0

### PR DESCRIPTION
**Pull request by pedrommcarrasco**
## What was done?
* Re-Developed Constrictor from the ground.
* Re-Developed every single test and added snapshot testing

## Why?
This was mainly done because its learning curve was way too high, as well as, adding the "update constraint feature" was way too hard in its previous state.

## Code Sample
I strongly advise you to read the new **[README.md](https://github.com/pedrommcarrasco/Constrictor/tree/Revamp)**
Still here are some examples
### New Features
It's now possible to easily update constraints without tricky identifiers or filters. 
```swift
// Only have one constraint for an anchor?
label.constrictor.updateFirst(.top) { $0?.enable() }
// Have two constraints for an anchor but for different elements? Provide more details
label.constrictor.update(.bottom, to: imageView) { $0?.constant = 16 }
// Call UIView.animate(...) { view.layoutIfNeeded() } to animate changes
```
### Lost Features
* Lost safeArea's abstraction - **To do? No plans**
* Lost the `inParent`/`toParent` sugar - **To do? Maybe in the future**

#### Example 1
**4.1.0 (Before)**
```swift
guide.constrictEdges(to: contentView, with: .all(8))
```
**5.0.0 (Now)**
```swift
guide.constrictor
    .edge(to: contentView, with: 8)
```
#### Example 2
**4.1.0 (Before)**
```swift
logoImageView
    .constrict(to: guide, attributes: .top, .centerX, .width)
    .constrict(.height, to: guide, attribute: .width, multiplyBy: 0.5)
```
**5.0.0 (Now)**
```swift
logoImageView.constrictor
    .top(to: guide)
    .centerX(to: guide)
    .width(to: guide)
    .height(to: guide, .width, multiplyBy: 0.5)
```
#### Example 3
**4.1.0 (Before)**
```swift
stackView
    .constrictCenterInParent()
    .constrict(.width, with: 100)
```
**5.0.0 (Now)**
```swift
stackView.constrictor
    .center(in: self)
    .width(to: 100)
```

### CODE REVIEW

File: Constrictor/Constrictor/Anchors/Anchor.swift [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        case .height: return .height
+        }
+    }
+}
+
+// MARK: - Functions
+extension Anchor {
+    
+    func constraints(for constraints: Constraints) -> [NSLayoutConstraint] {
+        switch self{
```
Comment (NSMyself) [inline Constrictor/Constrictor/Anchors/Anchor.swift]: please add a space between "self{" 😄

File: Constrictor/Constrictor/Anchors/AnchorDimension.swift [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    func anchor(for anchorable: Anchorable) -> NSLayoutDimension {
+        switch self {
+        case .width: return anchorable.widthAnchor
+        case .height: return anchorable.heightAnchor
+        }
+    }
+    
+    func constraints(for constraints: Constraints) -> [NSLayoutConstraint] {
+        switch self{
```
Comment (NSMyself) [inline Constrictor/Constrictor/Anchors/AnchorDimension.swift]: Same

**Review by anafpf — DISMISSED** [review]

**Comment by NSMyself** [comment]
Great job 👍

**Review by NSMyself — DISMISSED** [review]

File: Constrictor/Constrictor/Anchors/Anchor.swift (hunk 1 above)
Comment (pedrommcarrasco) [reply Constrictor/Constrictor/Anchors/Anchor.swift]: Nice catch! 👍

**Review by anafpf — APPROVED** [review]

**Review by NSMyself — APPROVED** [review]

**Pull request merged**